ethcore/src/block.rs

```diff
@@ -222,6 +222,7 @@ impl<'x> OpenBlock<'x> {
 		self.block.header.set_timestamp(header.timestamp());
 		self.block.header.set_uncles_hash(*header.uncles_hash());
 		self.block.header.set_transactions_root(*header.transactions_root());
+		self.block.header.set_seal(header.seal().to_vec());
```

Comment:
The fix for #11445 is here: without this the `LockedBlock` built by `enact()` lacks the seal and this causes the state root to be wrong.

Reply:
This doesn't seem right for me though. The idea with block types is that you go from:
`OpenBlock -> ClosedBlock -> LockedBlock -> SealedBlock`
and only the last type is guaranteed to have the seal.

How come the `seal` is included in the state root anyway? It's two separate things and really I don't see a way how `seal` affects the `state`?

Reply:
Added comment + ref to this PR. It still feels a bit dirty to do this here and I'm open to alternatives.

Reply:
question: is this not called from `Miner::seal_and_import_block_internally` code path? It seems like `Miner::prepare_block` doesn't call `OpenBlock::new`, correct?

Reply:
Not sure I understand what you mean, but the reverse call-graph for `populate_from()` looks like this:

```
populate_from() is called by…
	enact() is called by…
		enact_verified() is called by…
			check_and_lock_block() is called by…
				import_verified_blocks() is called by… (in impl Importer)
					import_verified_blocks() is called by… (in impl ImportBlock for Client)
						message() is called by… (ClientIoMessage::BlockVerified)
						flush_queue()
```

In other words, `populate_from()` is only called on the block verification path, not for block producing.

Reply:
cool, yeah that's what I meant.